addon/mixins/rest-serializer.js

```diff
@@ -0,0 +1,23 @@
+import Ember from 'ember';
+import baseSerializerMixin from 'ember-data-partial-model/mixins/base-serializer';
+const { Mixin } = Ember;
+
+export default Mixin.create(baseSerializerMixin, {
+  // Keep it here just to indicate the same interace
+  // used in jsonapi serialier
+  serializeBelongsTo: function(/*snapshot, json, relationship */) {
+    return this._super(...arguments);
+  },
+
+  _normalizePartialRelationship: function(modelClass, resourceHash, descriptor) {
+    resourceHash[descriptor.key] = resourceHash.id;
+  },
+
+  _copyAttributesFromPartialToParent: function(serializedHash, partialHash) {
+    Ember.merge(serializedHash, partialHash);
+  },
+
+  _modifyHashBasedOnDescriptorKey: function(serializedHash, descriptorKey) {
+    delete serializedHash[descriptorKey];
```

Comment:
I think it should probably be
`delete serializedHash[this.keyForRelationship(descriptorKey, 'belongsTo')]` instead of 
`delete serializedHash[descriptorKey];`